.eslintrc.json

```diff
@@ -21,6 +21,7 @@
     },
     "rules": {
         "semi": ["warn", "always"],
+        "no-use-before-define": ["warn"],
```

Comment:
I think we should downgrade this rule even further, into non-existence. Robert C. Martin says in Clean Code that for readability "If one function calls another, they should be vertically close, and the caller should be above the callee, if at all possible." I tend to agree.

Reply:
If you get rid of it, is it worth reverting the changes in this commit? If we revert, we get closer to the Robert C. Martin ideal, but if not, we would get consistency.

Reply:
whilst i havent read this book, i am in agreement, its readability that becomes the primary concern if ordering of functions wont *break* anything. id rather see associated methods near one another rather than one of them being 100 lines away just because it uses a utility function or something like this. agreed, will disable this rule entirely